test/onnx/internal/fx/test_frontend_consistency.py

```diff
@@ -0,0 +1,157 @@
+# Owner(s): ["module: onnx"]
+
+"""Test consistency between the output values of fx.GraphModule produced by FX frontend
+and torch.nn.Modules.
+"""
+
+from __future__ import annotations
+
+import functools
+from typing import Sequence, Union
+
+import torch
+from torch import nn
+from torch.onnx._internal.fx import frontend
+from torch.testing._internal import common_device_type, common_modules, common_utils
+from torch.utils import _pytree as pytree
+
+_FrontendType = Union[frontend.FxFrontend, frontend.FxFrontendUnpackKwargs]
+FX_FRONTENDS: Sequence[_FrontendType] = [
+    # TODO(bowbao): skip aotautograd, because it alters graph inputs/outputs.
+    # frontend.FxFrontendUnpackKwargs(frontend.AOTAutogradFrontend(dynamic=True)),
+    frontend.DynamoExport(tracing_mode="symbolic", aten_graph=True),
+    frontend.DynamoExport(tracing_mode="real", aten_graph=True),
+    frontend.FxFrontendUnpackKwargs(frontend.DynamoOptimize(dynamic=True)),
+    frontend.FxFrontendUnpackKwargs(frontend.MakeFx(tracing_mode="real")),
+    frontend.FxFrontendUnpackKwargs(frontend.DynamoOptimize(dynamic=False)),
+]
+
+# TODO: Organize test models into a separate file to be re-used by other tests.
+class MLPModel(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.fc0 = nn.Linear(8, 8, bias=True)
+        self.fc1 = nn.Linear(8, 4, bias=True)
+        self.fc2 = nn.Linear(4, 2, bias=True)
+        self.fc3 = nn.Linear(2, 2, bias=True)
+
+    def forward(self, tensor_x: torch.Tensor):
+        tensor_x = self.fc0(tensor_x)
+        tensor_x = torch.sigmoid(tensor_x)
+        tensor_x = self.fc1(tensor_x)
+        tensor_x = torch.sigmoid(tensor_x)
+        tensor_x = self.fc2(tensor_x)
+        tensor_x = torch.sigmoid(tensor_x)
+        output = self.fc3(tensor_x)
+        return output
+
+
+def module_inputs_mlp_model(
+    module_info, device, dtype, requires_grad, training, **kwargs
+):
+    make_input = functools.partial(
+        common_modules.make_tensor,
+        device=device,
+        dtype=dtype,
+        requires_grad=requires_grad,
+    )
+    module_inputs = [
+        common_modules.ModuleInput(
+            constructor_input=common_modules.FunctionInput(),
```

Comment:
What is `constructor_input` input? Why is `common_modules.FunctionInput()` assigned to it? Does it mean a `None`?

Reply:
Added doc in `Note [Add new module test cases]`.

It means the `__init__` of that module takes 0 arguments.